### PR TITLE
Port spline interpolation

### PR DESCRIPTION
**Pull request by cvasfi**
With this PR, spline interpolation package can be built with ROS2 foxy.

### CODE REVIEW

**Comment by cvasfi** [comment]
@mitsudome-r Shall i pick a reviewer randomly or is there a process defined for that?

File: common/math/spline_interpolation/CMakeLists.txt [hunk 1]
```diff
@@ -1,43 +1,14 @@
-cmake_minimum_required(VERSION 3.0.2)
+cmake_minimum_required(VERSION 3.5)
 project(spline_interpolation)
 
-add_compile_options(-std=c++14)
+if(NOT CMAKE_CXX_STANDARD)
+    set(CMAKE_CXX_STANDARD 14)
+endif()
```
Comment (nnmm) [inline common/math/spline_interpolation/CMakeLists.txt:6]: There's some more boilerplate we usually add:
```
if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
  add_compile_options(-Wno-unused-parameter -Wall -Wextra -Wpedantic)
endif()
```
I think Fred has even more in his newest porting doc PR

Comment (fred-apex-ai) [reply common/math/spline_interpolation/CMakeLists.txt:6]: @nnmm yes, that's true but we can make the standard enforcement stricter in a 2nd step.

@cvasfi please just add the warning flags, and while building temporarily add `-Werror` to make sure there are no  warnings left but don't commit `-Werror`

Comment (mitsudome-r) [reply common/math/spline_interpolation/CMakeLists.txt:6]: Looks like `Werror` doesn't show any error so I will merge this.

**Review by mitsudome-r — APPROVED** [review]

**Pull request merged**